### PR TITLE
Fix compilation in OSX Catalina with OMP enabled

### PR DESCRIPTION
**Pull request by SergioRAgostinho**
- Explicitly add linker flags from OMP

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -297,6 +297,9 @@ if(OPENMP_FOUND)
     else()
       message(WARNING "Delay loading flag for OpenMP DLL is invalid.")
     endif()
+  else()
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${OpenMP_CXX_LIBRARIES}")
+    set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${OpenMP_CXX_LIBRARIES}")
```
Comment (SergioRAgostinho) [inline CMakeLists.txt]: This line is causing the cmake failures.

**Comment by shrijitsingh99** [comment]
This problem managed to escape the macOS CI since OMP was not installed. We should add it the CI will help catch any future errors that may arise.

**Comment by SergioRAgostinho** [comment]
Go for it :+1: However, it's not worth to commit the change until we fix the problem.

**Comment by shrijitsingh99** [comment]
> Go for it 👍 However, it's not worth to commit the change until we fix the problem.

Yeah true, we could just add it to this PR since its only one line (`brew install libomp`) added to `build-macos.yml`.

**Comment by SergioRAgostinho** [comment]
I'm not gonna touch this for the time being, so feel free to take over the issue and signal the closing of this PR if you get yours ready.

**Comment by taketwo** [comment]
So unlike all other OSes, Catalina needs OMP to be explicitly linked? This needs to be propagated to downstream packages then :(

**Comment by SergioRAgostinho** [comment]
I haven't addressed the downstream part yet i.e.: it the project is compiled with OMP and requires linking flags, the assumption is that if the linking flags are not propagated to a downstream project it will have linking issues.

**Comment by kunaltyagi** [comment]
I created an issue to potentially track the issue. We can go ahead and merge this

**Pull request merged**